src/rooms/IPreRoomCreateHandler.ts

```diff
@@ -2,7 +2,14 @@ import { IRead } from '../accessors/index';
 import { IRoom } from './IRoom';
 
 export interface IPreRoomCreateHandler {
-    preRoomCreate(room: IRoom): IRoom;
+    /**
+     * Checks whether the handler actually shall execute.
+     * This method can basically implement a filter.
+     *
+     * @param room
+     * @param read
+     */
+    checkPreRoomCreate(room: IRoom, read: IRead): boolean;
 
-    isRoomApplicable(message: IRoom, read: IRead): boolean;
+    preRoomCreate(room: IRoom): IRoom;
```

Comment:
Yeah this one was not completed at all, was mostly just to add another interface to show a co-worker that a class could implement several interfaces in the same class, how it worked, and what the compiled version looked like.